Source/WebKit/WebProcess/WebStorage/StorageNamespaceImpl.cpp

```diff
@@ -42,19 +42,19 @@
 namespace WebKit {
 using namespace WebCore;
 
-static HashMap<StorageNamespaceImpl::Identifier, StorageNamespaceImpl*>& sessionStorageNamespaces()
+static HashMap<std::pair<StorageNamespaceImpl::Identifier, WebCore::SecurityOriginData>, StorageNamespaceImpl*>& sessionStorageNamespaces()
 {
-    static NeverDestroyed<HashMap<StorageNamespaceImpl::Identifier, StorageNamespaceImpl*>> map;
+    static NeverDestroyed<HashMap<std::pair<StorageNamespaceImpl::Identifier, WebCore::SecurityOriginData>, StorageNamespaceImpl*>> map;
     return map;
 }
 
-Ref<StorageNamespaceImpl> StorageNamespaceImpl::createSessionStorageNamespace(Identifier identifier, PageIdentifier pageID, unsigned quotaInBytes)
+Ref<StorageNamespaceImpl> StorageNamespaceImpl::createSessionStorageNamespace(Identifier identifier, PageIdentifier pageID, WebCore::SecurityOrigin& topLevelOrigin, unsigned quotaInBytes)
 {
     // The identifier of a session storage namespace is the WebPageProxyIdentifier. It is possible we have several WebPage objects in a single process for the same
     // WebPageProxyIdentifier and these need to share the same namespace instance so we know where to route the IPC to.
-    if (auto* existingNamespace = sessionStorageNamespaces().get(identifier))
+    if (auto* existingNamespace = sessionStorageNamespaces().get(std::make_pair(identifier, topLevelOrigin.data())))
```

Comment:
nit: Can we use initializer lists instead of std::make_pair?  The latter is a bit verbose